utils/python/CIME/case_setup.py

```diff
@@ -190,6 +190,7 @@ def _case_setup_impl(case, caseroot, clean=False, test_mode=False, reset=False):
                    "it is not possible to run with OpenMP if using the NAG Fortran compiler")
             cost_pes = env_mach_pes.get_cost_pes(pestot, thread_count, machine=case.get_value("MACH"))
             case.set_value("COST_PES", cost_pes)
+            case.set_value("TOTAL_CORES", pestot * thread_count)
```

Comment:
I think that this is incorrect in some cases.   On SMT enabled systems we put two or four threads on a single core.    That's why I use the ratio PES_PER_NODE/MAX_TASKS_PER_NODE